pkg/operator/statefulset.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"fmt"
+	"net/url"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+const configReloaderPort = 8080
+
+// CreateConfigReloader returns the definition of the config-reloader container.
+func CreateConfigReloader(
+	config ReloaderConfig,
+	reloadURL url.URL,
+	listenLocal bool,
+	localHost string,
+	logFormat, logLevel string,
+	additionalArgs []string,
+	volumeMounts []v1.VolumeMount,
+) v1.Container {
+	var (
+		ports []v1.ContainerPort
+		args  = make([]string, 0, len(additionalArgs))
+	)
+
+	var listenAddress string
+	if listenLocal {
+		listenAddress = localHost
+	} else {
+		ports = append(
+			ports,
+			v1.ContainerPort{
+				Name:          "web",
```

Comment:
nit: Can this port be named differently? We are using the same name for prometheus pod and this will cause pod to have two ports which are named the same. It might be a bit confusing, especially when it comes to writing a ServiceMonitor.

Reply:
Good point!